trecs/models/recommender.py

```diff
@@ -332,9 +388,13 @@ def generate_recommendations(self, k=1, item_indices=None):
                 )
         if k == 0:
             return np.array([]).reshape((self.num_users, 0)).astype(int)
+        # convert to dense because scipy does not yet support argsort - consider
+        # implementing our own fast sparse version? see
```

Comment:
Or our own `argsort` on top of scipy's sparse? Is that an option?

Reply:
Hmm, I don't know what you mean, do you mind expanding on that? The way I'm interpreting your comment is writing our own `argsort` method that implements the same functionality as `numpy.argsort` but works with `scipy` matrices. We could do this but to do it efficiently / fast seems like a lot of potential time consumption, so I didn't attempt it for this PR

Reply:
Yes,`argsort` would definitely be too much to implement for this PR and we should only think about it after the first release. I think I misunderstood your comment thinking you meant you wanted to implement our own version of sparse matrices from scratch instead of using the scipy implementation. Now I understand that's not the case and we actually meant the same thing.